### ISSUE TITLE
tests test_empty_string{1,2} fail under py3 in suite of -2.4.3

### ISSUE THREAD

**Issue by idella** [pydata/numexpr#177]
all pass under python2.7

```bash
............................................EE......
======================================================================
ERROR: test_empty_string1 (numexpr.tests.test_numexpr.test_strings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/tests/test_numexpr.py", line 725, in test_empty_string1
    res = evaluate("(a == '') & (b == 'pepe2')")
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/necompiler.py", line 756, in evaluate
    signature = [(name, getType(arg)) for (name, arg) in zip(names, arguments)]
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/necompiler.py", line 756, in <listcomp>
    signature = [(name, getType(arg)) for (name, arg) in zip(names, arguments)]
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/necompiler.py", line 654, in getType
    raise ValueError("unknown type %s" % a.dtype.name)
ValueError: unknown type str128

======================================================================
ERROR: test_empty_string2 (numexpr.tests.test_numexpr.test_strings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/tests/test_numexpr.py", line 733, in test_empty_string2
    res = evaluate("(a == '') & (b == 'pepe2')")
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/necompiler.py", line 756, in evaluate
    signature = [(name, getType(arg)) for (name, arg) in zip(names, arguments)]
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/necompiler.py", line 756, in <listcomp>
    signature = [(name, getType(arg)) for (name, arg) in zip(names, arguments)]
  File "/mnt/gen2/TmpDir/portage/dev-python/numexpr-2.4.3/work/numexpr-2.4.3-python3_4/lib/numexpr/necompiler.py", line 654, in getType
    raise ValueError("unknown type %s" % a.dtype.name)
ValueError: unknown type str128

----------------------------------------------------------------------
Ran 5433 tests in 9.672s

FAILED (errors=2)
```

python versions 3.3.5 and 2.4.3
numpy version is 1.9.2.
These appear to be the old 'py2 vs py3' string mismatch which I thought was long fixed, but that's just a glancing comment.
Do you require any further information?

**Comment by groutr** [comment]
I can confirm the same issues when building numexpr 2.4.3 for Python 3.3/3.4

**Comment by ffernand** [comment]
I can confirm that I have the same error.

* Python 3.4.3 (installed via pyenv)
* numexpr 2.4.3 (installed via PyPI)
* Ran... python -c "import numexpr; numexpr.test()"

**Comment by jmwoloso** [comment]
Got the same error.

Python 3.4.3 via a virtualenv
Built with Intel MKL from source
numexpr 2.4.3

Any movement on this?

**Comment by avalentino** [comment]
Fixed by PR #182

**Comment by jmwoloso** [comment]
Thank you very much for the update!
